delfin/drivers/ibm/ds8k/rest_handler.py

```diff
@@ -0,0 +1,129 @@
+        res = self.do_call(url, data, method)
+        if auth_key:
+            self.session.headers['X-Auth-Token'] = auth_key
+        return res
+
+    def call(self, url, data=None, method=None):
+        try:
+            res = self.call_with_token(url, data, method)
+            if res.status_code == 401:
+                LOG.error("Failed to get token,status_code:%s,error_mesg:%s" %
+                          (res.status_code, res.text))
+                self.login()
+                res = self.call_with_token(url, data, method)
+            elif res.status_code == 503:
+                raise exception.InvalidResults(res.text)
+            return res
+        except Exception as e:
+            LOG.error("Method:%s,url:%s failed: %s" % (method, url,
+                                                       six.text_type(e)))
+            raise e
+
+    def get_rest_info(self, url, data=None, method='GET'):
+        result_json = None
+        res = self.call(url, data, method)
+        if res.status_code == 200:
+            result_json = res.json()
+        return result_json
+
+    def login(self):
+        try:
+            data = {
+                'request': {
+                    'params': {
+                        "username": self.rest_username,
+                        "password": cryptor.decode(self.rest_password)
+                    }
+                }
+            }
+            with self.session_lock:
+                if self.session is None:
+                    self.init_http_head()
+                res = self.call_with_token(
+                    RestHandler.REST_TOKEN_URL, data, 'POST')
+                if res.status_code == 200:
+                    result = res.json()
+                    self.session.headers['X-Auth-Token'] = \
+                        cryptor.encode(result.get('token').get('token'))
+                else:
+                    LOG.error("Login error. URL: %(url)s，Reason: %(reason)s.",
+                              {"url": RestHandler.REST_TOKEN_URL,
+                               "reason": res.text})
+                    if 'Authentication has failed' in res.text:
+                        raise exception.InvalidUsernameOrPassword()
+                    else:
+                        raise exception.StorageBackendException(res.text)
+        except Exception as e:
+            LOG.error("Login error: %s", six.text_type(e))
+            raise e
+        finally:
+            data = None
+
+    def logout(self):
+        try:
+            if self.san_address:
+                self.call(RestHandler.REST_TOKEN_URL, method='DELETE')
+            if self.session:
+                self.session.close()
+        except Exception as e:
+            err_msg = "Logout error: %s" % (six.text_type(e))
+            LOG.error(err_msg)
+            raise e
+
+    def get_storage(self):
+        result_json = self.get_rest_info('/api/v1/systems')
+        return result_json
```

Comment:
In my opinion, API string `/api/v1/systems` should defined as a constant, and this fucntion should be deleted, use `system_info = self.rest_handler.get_rest_info('/api/v1/systems')`, because there is nothging to do for the reuslt fo this API. And same reason for other functions.

Reply:
done